src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Extensions/SyntaxNodeExtensions.vb

```diff
@@ -998,57 +999,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
         End Function
 
         ''' <summary>
-        ''' Given an expression within a tree of <see cref="ConditionalAccessExpressionSyntax"/>s, 
-        ''' finds the <see cref="ConditionalAccessExpressionSyntax"/> that it is part of.
+        ''' <see cref="ISyntaxFacts.GetRootConditionalAccessExpression"/>
         ''' </summary>
-        ''' <param name="node"></param>
-        ''' <returns></returns>
         <Extension>
-        Friend Function GetParentConditionalAccessExpression(node As ExpressionSyntax) As ConditionalAccessExpressionSyntax
-            Dim access As SyntaxNode = node
-            Dim parent As SyntaxNode = access.Parent
-
-            While parent IsNot Nothing
-                Select Case parent.Kind
-                    Case SyntaxKind.DictionaryAccessExpression,
-                         SyntaxKind.SimpleMemberAccessExpression
-
-                        If DirectCast(parent, MemberAccessExpressionSyntax).Expression IsNot access Then
-                            Return Nothing
-                        End If
-
-                    Case SyntaxKind.XmlElementAccessExpression,
-                         SyntaxKind.XmlDescendantAccessExpression,
-                         SyntaxKind.XmlAttributeAccessExpression
-
-                        If DirectCast(parent, XmlMemberAccessExpressionSyntax).Base IsNot access Then
-                            Return Nothing
-                        End If
-
-                    Case SyntaxKind.InvocationExpression
-
-                        If DirectCast(parent, InvocationExpressionSyntax).Expression IsNot access Then
-                            Return Nothing
-                        End If
-
-                    Case SyntaxKind.ConditionalAccessExpression
-
-                        Dim conditional = DirectCast(parent, ConditionalAccessExpressionSyntax)
-                        If conditional.WhenNotNull Is access Then
-                            Return conditional
-                        ElseIf conditional.Expression IsNot access Then
-                            Return Nothing
-                        End If
+        Friend Function GetRootConditionalAccessExpression(node As ExpressionSyntax) As ConditionalAccessExpressionSyntax
+            ' Walk upwards based on the grammar/parser rules around ?. expressions (can be seen in
+            ' ParseExpression.vb (.ParsePostFixExpression).
+            '
+            ' These are the parts of the expression that the ?... expression can end with.  Specifically
+            '
+            '  1.      x?.y.M()             // invocation
+            '  2.      x?.y and x?.y.z      // member access
+            '  4.      x?!y                 // dictionary access
+            '  5.      x?.y<...>            // xml access
```

Comment:
i added the // comments here to the C# side in anotehr commit as i think they're helpful.